src/main/java/com/timgroup/statsd/StatsDSender.java

```diff
@@ -45,7 +46,15 @@
         this.address = addressLookup.call();
         this.clientChannel = clientChannel;
 
-        this.executor = Executors.newFixedThreadPool(workers);
+        this.executor = Executors.newFixedThreadPool(workers, new ThreadFactory() {
+            final ThreadFactory delegate = Executors.defaultThreadFactory();
+            @Override public Thread newThread(final Runnable runnable) {
+                final Thread result = delegate.newThread(runnable);
+                result.setName("StatsD-Sender-" + result.getName());
+                result.setDaemon(true);
```

Comment:
I'm kind of torn about making this a daemon thread or not. On one hand this is an IO intensive thread, and we might also want to make sure it shuts down in orderly fashion, on the other hand this runs next to the application and one could argue shouldn't really be a user thread....

Reply:
I don't have a strong option about it as well. But I thought it might be a good idea to make it a daemon since `com.timgroup.statsd.NonBlockingStatsDClient` is also creating daemon threads.